pkg/services/publicdashboards/service/query.go

```diff
@@ -239,9 +239,22 @@ func getUniqueDashboardDatasourceUids(dashboard *simplejson.Json) []string {
 func groupQueriesByPanelId(dashboard *simplejson.Json) map[int64][]*simplejson.Json {
 	result := make(map[int64][]*simplejson.Json)
 
-	for _, panelObj := range dashboard.Get("panels").MustArray() {
+	extractQueriesFromPanels(dashboard.Get("panels").MustArray(), result)
+
+	return result
+}
+
+func extractQueriesFromPanels(panels []interface{}, result map[int64][]*simplejson.Json) {
+	for _, panelObj := range panels {
 		panel := simplejson.NewFromAny(panelObj)
 
+		// if the panel is a row and it is collapsed, get the queries from the panels inside the row
+		if panel.Get("type").MustString() == "row" && panel.Get("collapsed").MustBool() == true {
```

Comment:
isn't there a way to start the view with a panel row expanded? I mean, not collapsed, so this condition won't work.
Based on pubdash page component, I don't think so.
Has the dashboard that behavior? Should we add it?

Reply:
What do you mean? You can save a dashboard with rows collapsed or expanded. Then, the public dashboard will reflect that behavior, so yes, it is possible.

Reply:
I mean, is it possible to pass a query param and start with the dashboard not collapsed, so `panel.Get("collapsed").MustBool() == true` will be false?
I just checked and it seems that's not possible